src/PostgREST/App.hs

```diff
@@ -227,33 +227,43 @@ app dbStructure conf apiRequest =
               let acceptH = (hAllow, if tableInsertable table then "GET,POST,PATCH,DELETE" else "GET") in
               return $ responseLBS status200 [allOrigins, acceptH] ""
 
-        (ActionInvoke _isReadOnly, TargetProc qi, payload) ->
-          let proc = M.lookup (qiName qi) allProcs
+        (ActionInvoke _, TargetProc qi, Just (PayloadJSON payload pType pKeys)) ->
+          let paramsAsSingleObject = iPreferSingleObjectParameter apiRequest
+              procs = M.lookup (qiName qi) allProcs
+              proc = join $
+                -- Handle overloaded functions case
+                (case length <$> procs of
```

Comment:
It will work but the idea was to immediately get the `ProcDescription` if the function is not overloaded(majority of cases) and avoid the extra steps, creating a Set and comparing two Sets, do you think that is not worth to do that?

Reply:
I see what you mean. Well maybe update the comment to explain that we're doing an optimization for the length=1 case.